src/types/actions.ts

```diff
@@ -6,16 +6,15 @@
  *
  * Actions must have a `type` field that indicates the type of action being
  * performed. Types can be defined as constants and imported from another
- * module. It's better to use strings for `type` than Symbols because strings
- * are serializable.
+ * module. These must be strings, as strings are serializable.
  *
  * Other than `type`, the structure of an action object is really up to you.
  * If you're interested, check out Flux Standard Action for recommendations on
  * how actions should be constructed.
  *
  * @template T the type of the action's `type` tag.
  */
-export interface Action<T = any> {
+export interface Action<T extends string = string> {
```

Comment:
it can be useful to have a specific literal string for your action type - for example, if your middleware implements a different call signature when you dispatch a specific type

`Middleware<{ (action: Action<'middleware/myCustom'>): MyReturn }>`

Reply:
Ah, of course. Didn't think about that.